src/SonarScanner.MSBuild.Common/ConsoleLogger.cs

```diff
@@ -137,6 +138,31 @@ public void WriteUIWarnings(string outputFolder)
         }
     }
 
+    public void AddTelemetryMessage(string key, string value) =>
+        telemetryMessages.Add(new(key, value));
+
+    public void WriteTelemetry(string outputFolder)
+    {
+        var telemetryMessagesJson = new StringBuilder();
+        foreach (var message in telemetryMessages)
+        {
+            var entry = new JObject();
+            entry[message.Key] = message.Value;
+            telemetryMessagesJson.AppendLine(entry.ToString(Formatting.None));
+        }
+
+        var path = Path.Combine(outputFolder, FileConstants.TelemetryFileName);
+        var telemetry = telemetryMessagesJson.ToString();
+        if (!fileWrapper.Exists(path))
+        {
+            fileWrapper.WriteAllText(path, telemetry);
+        }
+        else
+        {
+            fileWrapper.AppendAllText(path, telemetry);
+        }
```

Comment:
[File.AppendAllText](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.appendalltext?view=net-8.0)

> Appends the specified string to the file, creating the file if it does not already exist.


```suggestion
        fileWrapper.AppendAllText(path, telemetry);
```